### PR TITLE
fix payee name selection based on the amount

### PR DESCRIPTION
**Pull request by matt-fidd**
Fixes actualbudget/actual#3216

### CODE REVIEW

File: src/util/payee-name.js [hunk 1]
```diff
@@ -5,12 +5,13 @@ function formatPayeeIban(iban) {
 }
 
 export const formatPayeeName = (trans) => {
+  const amount = trans.amount || trans.transactionAmount.amount;
```
Comment (MatissJanis) [inline src/util/payee-name.js:8]: **💬 suggestion:** ‏do we expect `trans.amount` to be `undefined`? If yes - then using `??` would be more appropriate.

```suggestion
  const amount = trans.amount ?? trans.transactionAmount.amount;
```

Comment (MatissJanis) [reply src/util/payee-name.js:8]:
```javascript
console.log(0 ?? 12); // 0
console.log(0 || 12); // 12

console.log(undefined ?? 12); // 12
console.log(undefined || 12); // 12
```

Comment (matt-fidd) [reply src/util/payee-name.js]: Fixed, turns out `trans.amount` is never set - I missed it when I ported the util over from actual.

**Review by MatissJanis — APPROVED** [review]

**Comment by Soulplayer** [comment]
Bug remains. See https://github.com/actualbudget/actual/issues/3216#issuecomment-2299358407

**Pull request merged**